File: src/main/kotlin/com/nazarethlabs/codex/service/note/ExportNotesService.kt
package com.nazarethlabs.codex.service.note

import com.nazarethlabs.codex.dto.Note
import com.nazarethlabs.codex.helper.ZipHelper
import com.nazarethlabs.codex.service.sentry.SentryEventHelper
import java.io.File

class ExportNotesService {
    fun export(
        notes: List<Note>,
        outputPath: String,
    ): File {
        val filePaths = notes.map { it.filePath }
        val zipPath = ZipHelper.createZipFromFiles(filePaths, outputPath)
        SentryEventHelper.captureEvent("note.exported")
        return File(zipPath)
    }
}

File: vscode/src/service/ExportNotesService.ts
import * as path from 'path';
import { Note } from '../dto/Note';
import { NoteRepository } from '../repository/NoteRepository';
import { FileHelper } from '../helper/FileHelper';
import { DialogHelper } from '../helper/DialogHelper';
import { ZipHelper } from '../helper/ZipHelper';

export class ExportNotesService {
  private readonly repository: NoteRepository;

  constructor(repository: NoteRepository) {
    this.repository = repository;
  }

  async exportAll(): Promise<void> {
    const saveUri = await DialogHelper.showSaveDialog({
      filters: { 'ZIP Archive': ['zip'] },
      defaultUri: undefined,
    });

    if (!saveUri) {
      return;
    }

    const notes = this.repository.getAllNotes();
    const filePaths = notes
      .map((n) => n.filePath)
      .filter((fp) => !!fp);

    const outputPath = saveUri.fsPath.endsWith('.zip') ? saveUri.fsPath : `${saveUri.fsPath}.zip`;

    await ZipHelper.createZipFromFiles(filePaths, outputPath);
    DialogHelper.showInfo(`Exported ${filePaths.length} notes to ${path.basename(outputPath)}`);
  }
}
